arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,79 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor{arduinoRuntime, smartcarlib::pins::v2::leftMotorPins};
+BrushedMotor rightMotor{arduinoRuntime, smartcarlib::pins::v2::rightMotorPins};
+DifferentialControl control{leftMotor, rightMotor};
+
+SimpleCar car(control);
+
+const int triggerPin           = 6; // D6
+const int echoPin              = 7; // D7
+const unsigned int maxDistance = 300;
+const int fspeed = 50;
+const int bspeed = -50;
+const int rdegrees = 75;
+const int ldegrees = -75;
+SR04 front{arduinoRuntime, triggerPin, echoPin, maxDistance};
+
+void setup()
+{
+  // Setup port to receive input
+  Serial.begin(9600);
+  //Setup car to move 50% of its speed
+  car.setSpeed(fspeed);
+
+
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void loop()
+{
+ detectObstacle();
+ handleInput();
+#ifdef __SMCE__
+  // Avoid over-using the CPU if we are running in the emulator
+  delay(1000);
```

Comment:
Isn't waiting one second a bit excessive? 😅
Also, if you have the latest smce you don't need that anymore.